server/model/taggedBailViewModel.ts

```diff
@@ -0,0 +1,77 @@
+import { PrisonApiOffenderSentenceAndOffences, PrisonApiPrisoner } from '../@types/prisonApi/prisonClientTypes'
+import { Adjustment } from '../@types/adjustments/adjustmentsTypes'
+import { AdjustmentType } from './adjustmentTypes'
+
+export default class TaggedBailViewModel {
+  constructor(
+    public prisonerDetail: PrisonApiPrisoner,
+    public adjustments: Adjustment[],
+    public adjustmentType: AdjustmentType,
+    public sentencesAndOffences: PrisonApiOffenderSentenceAndOffences[],
+  ) {}
+
+  public backlink(): string {
+    return `/${this.prisonerDetail.offenderNo}`
+  }
+
+  public columnHeadings() {
+    return [{ text: 'Court name' }, { text: 'Case reference' }, { text: 'Days' }, { text: 'Actions' }]
+  }
+
+  public rows() {
+    return this.adjustments.map(it => {
+      return [
+        { text: this.getCourtName(it.taggedBail.caseSequence) },
+        { text: this.getCaseReference(it.taggedBail.caseSequence) },
+        { text: it.days || it.daysBetween || it.effectiveDays },
+        this.actionCell(it),
+      ]
+    })
+  }
+
+  public table() {
+    return {
+      head: this.columnHeadings(),
+      rows: this.rows().concat(this.totalRow()),
+      attributes: { 'data-qa': 'view-table' },
+    }
+  }
+
+  public totalRow() {
+    const total = this.adjustments.map(it => it.days || it.daysBetween || it.effectiveDays).reduce((a, b) => a + b, 0)
+    return [[{ html: '<b>Total days</b>' }, { html: '' }, { html: `<b>${total}</b>` }, { text: '' }]]
+  }
+
+  private getCourtName(caseSequence: number): string {
+    const sentenceAndOffence = this.sentencesAndOffences.find(it => it.caseSequence === caseSequence)
```

Comment:
So there can be multiple sentences on the same court case (Will have the same case sequence).

Have a look at how we select the sentence by sentence date in the add journey.
https://github.com/ministryofjustice/hmpps-adjustments/blob/main/server/model/taggedBailSelectCaseModel.ts#L32

Reply:
I've moved some logic out of the `taggedBailSelectCaseModel.ts` into the `utils.ts` so that they can be used by the `taggedBailViewModel.ts` too. This way they're both using the same logic 👍